repo/repo.go

```diff
@@ -25,6 +25,8 @@ type Repo interface {
 	// SetAPIAddr sets the API address in the repo.
 	SetAPIAddr(addr string) error
 
+	Self() Repo
```

Comment:
the `node.Repo` should either be an fsrepo or a mock repo. It should work with a simple type assertion

> In this commit `node.Repo.(*fsrepo.FSRepo)` does not work as node.Repo is `repo.ref`

I'm not sure what you mean by this. what is repo.ref?